src/transactions-manager.ts

```diff
@@ -1,16 +1,22 @@
-import { enqueue } from "./sync-queue";
-import { type Transaction } from "./transaction";
+import type { Change, Transaction } from "./transaction";
+
+export type TransactionCallback = (
+  transactionId: string,
+  changes: Array<Change>
+) => void;
 
 export class TransactionsManager {
   max = 100;
   currentStack: Array<Transaction> = [];
   undoneStack: Array<Transaction> = [];
 
+  constructor(private callback: TransactionCallback) {}
```

Comment:
weird, callback is not being referenced on the instance

Reply:
What do you mean?

Reply:
wait, is this syntax with `private callback` automatically referencing callback on the instance????

Reply:
Yes, was in ts since early days

Reply:
its a bit confusing since a private thing should be only accessible through private instance properties, I guess there was no private properties in js classes back then, in which case I would not use it and would actually use everything that is supposed to be private member as a real private method

I can see here that I forgot to use it for other properties as well